Change executor to match previous dist output

apinkert opened the pull request:
For [RHCLOUD-31842](https://issues.redhat.com/browse/RHCLOUD-31842). @Hyperkid123 I _think_ this should get us back to where we were. I'm able to match the output of https://www.npmjs.com/package/@redhat-cloud-services/rbac-client/v/1.2.13?activeTab=code if I delete the current `dist/` in a client, make a change to the `README`, and then build. Worst case it also appends the contents of the previous `dist` into the main new `dist`. Next step for this is to make `README` changes to all the clients with this new executor to revert to the previous output in a minor bump. Then, I'll open a tech debt story to revert all these changes in a major bump for all clients, forcing the normal `dist` output we have currently here https://www.npmjs.com/package/@redhat-cloud-services/rbac-client?activeTab=code

Hyperkid123 reviewed (changes requested):
If there is supposed to a dist directory, you also have to change the `main`, `module` and `typings` entries in the `package.json`

Now it is configured to
```
  "main": "./index.js",
  "module": "./esm/index.js",
  "typings": "./index.d.ts",
```

That will result in errors while using relative imports.

I also see that when I build the packages locally I see a structure like
```
dist
-- dist
--- api.tx
-api.tx
```
This is not the right way of doing the build. We just copy files to a nested directory. The problem is that npm publish is not called from the actual project root, but from the `dist` itself.

Its partially caused by the fact that the NX was reconfigured to include the build output next to the package source files, instead of having a top level `dist` output directory with all packages (nx default). That makes it goofy when it comes the actual directory structure of the outputs. Can we see the nx default instead? Have a common output dir that is structured based on the packages so we can play customize the package directories without having to mess with the source code structure?

Hyperkid123 commented on packages/rbac/project.json:
```diff
@@ -7,15 +7,23 @@
+      "options": {
+          "commands": [
+            "mkdir -p dist/rbac/dist dist/rbac/doc dist/rbac/.openapi-generator",
+            "cp packages/rbac/README.md packages/rbac/package.json packages/rbac/*.ts packages/rbac/*.json packages/rbac/LICENSE packages/rbac/.openapi-generator-ignore packages/rbac/postProcess.sh dist/rbac",
```
You are copying way too many files.

from this line you only need:
- packages/rbac/README.md
- packages/rbac/package.json
- packages/rbac/LICENSE
- (missing) packages/rbac/CHANGELOG.md

Hyperkid123 commented on packages/rbac/project.json:
```diff
@@ -7,15 +7,23 @@
+            "cp packages/rbac/README.md packages/rbac/package.json packages/rbac/*.ts packages/rbac/*.json packages/rbac/LICENSE packages/rbac/.openapi-generator-ignore packages/rbac/postProcess.sh dist/rbac",
+            "cp packages/rbac/dist/* dist/rbac/dist",
+            "cp -r packages/rbac/doc/* dist/rbac/doc",
+            "cp packages/rbac/.openapi-generator/* dist/rbac/.openapi-generator"
```
You don't want any of these files in your NPM package

Hyperkid123 commented on packages/rbac/project.json:
```diff
@@ -7,15 +7,23 @@
+      "executor": "nx:run-commands",
+      "options": {
+          "commands": [
+            "mkdir -p dist/rbac/dist dist/rbac/doc dist/rbac/.openapi-generator",
```
We don't need this directory `dist/rbac/.openapi-generator`

Hyperkid123 reviewed (commented):
The dist now has a bit too much in it, but the output seems almost there. You essentially only want:

- the compiled JS (which is under the dist/rbac/dist)
- the package.json, license, readme, and other openousrci things
- the docs dir is handy but not required in NPM bundle, its important for GH though
- and all the `*.d.ts` files from the dist output (which you have)

Also, be sure to properly set the package.json metadata:
- main
- typings
- module (should be removed nov as we only have a single build output)

apinkert commented:
@Hyperkid123 cool beans, I've updated the rest of the clients to match what we've done with `rbac-client`. All clients should now correctly build and copy files over to a top level `dist` with all clients pointing at their respective `dist/index` files. I've added the `copyMetadata` step to GH actions so it `should` build fine from here? I might need to to make small changes to the clients themselves to trigger the new releases after this PR.

Hyperkid123 commented:
I think the build should pick up the changes
